io-engine/src/bdev/nexus/nexus_io.rs

```diff
@@ -498,15 +539,24 @@ impl<'n> NexusBio<'n> {
 
             self.channel_mut().disconnect_device(&device);
 
-            self.retire_device(
+            self.fault_device(
                 &device,
                 IoCompletionStatus::IoSubmissionError(
-                    IoSubmissionFailure::Read,
+                    IoSubmissionFailure::Write,
                 ),
             );
         }
 
-        // partial submission
+        // Log all write-like operation in the rebuild logs, if any exist.
+        self.channel().for_each_rebuild_log(|log| {
```

Comment:
looks like submit_all function is called for  the following IoType,
           IoType::Write
            | IoType::WriteZeros
            | IoType::Reset
            | IoType::Unmap
Hence we should log the IoType as well along with the updated or not? Even if we don't support unmap, for now, we should have the provision in the design.

Reply:
I do pass IoType to the rebuild log, if you're asking that.